File: sample-chat-obj-c/sample-chat/Components/ChatScreen/Views/ChatCollectionView/ChatDateCell/ChatDateCell.m
//
//  ChatDateCell.m
//  sample-chat
//
//  Created by Injoit on 08.03.2020.
//  Copyright © 2020 Quickblox. All rights reserved.
//

#import "ChatDateCell.h"
#import "UIView+Chat.h"

@implementation ChatDateCell

- (void)awakeFromNib {
    [super awakeFromNib];
    
    [self.dateBackgroundView setRoundViewWithCornerRadius:11.0f];
    self.dateLabel.translatesAutoresizingMaskIntoConstraints = NO;
    self.dateBackgroundView.translatesAutoresizingMaskIntoConstraints = NO;
    [self.dateLabel.centerXAnchor constraintEqualToAnchor:self.centerXAnchor].active = YES;
    [self.dateLabel.centerYAnchor constraintEqualToAnchor:self.centerYAnchor].active = YES;
    [self.dateLabel.heightAnchor constraintEqualToConstant:15.0f].active = YES;
    [self.dateBackgroundView.topAnchor constraintEqualToAnchor:self.dateLabel.topAnchor constant:-3.0f].active = YES;
    [self.dateBackgroundView.leadingAnchor constraintEqualToAnchor:self.dateLabel.leadingAnchor constant:-16.0f].active = YES;
    [self.dateBackgroundView.bottomAnchor constraintEqualToAnchor:self.dateLabel.bottomAnchor constant:3.0f].active = YES;
    [self.dateBackgroundView.trailingAnchor constraintEqualToAnchor:self.dateLabel.trailingAnchor constant:15.0f].active = YES;
}

+ (ChatCellLayoutModel)layoutModel {
    ChatCellLayoutModel defaultLayoutModel = [super layoutModel];
    defaultLayoutModel.containerInsets = UIEdgeInsetsMake(4, 10, 4, 10);
    defaultLayoutModel.avatarSize = CGSizeZero;
    defaultLayoutModel.topLabelHeight = 0.0f;
    defaultLayoutModel.timeLabelHeight = 0.0f;
    return defaultLayoutModel;
}

@end

File: sample-chat-swift/sample-chat-swift/Classes/Chat/CollectionView/CollectionCells/ChatDateCell/ChatDateCell.swift
//
//  ChatDateCell.swift
//  sample-chat-swift
//
//  Created by Injoit on 12/3/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

import UIKit

class ChatDateCell: ChatCell {
    @IBOutlet weak var dateBackgroundView: UIView!
    @IBOutlet weak var dateLabel: UILabel!
    
    override class func layoutModel() -> ChatCellLayoutModel {
        let containerInsets = UIEdgeInsets(top: 4.0, left: 10.0, bottom: 4.0, right: 10.0)
        var defaultLayoutModel = super.layoutModel()
        defaultLayoutModel.containerInsets = containerInsets
        defaultLayoutModel.avatarSize = .zero
        defaultLayoutModel.topLabelHeight = 0.0
        defaultLayoutModel.timeLabelHeight = 0.0
        
        return defaultLayoutModel
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        dateBackgroundView.setRoundView(cornerRadius: 11)
        dateLabel.translatesAutoresizingMaskIntoConstraints = false
        dateBackgroundView.translatesAutoresizingMaskIntoConstraints = false
        let constraints = [
            dateLabel.centerXAnchor.constraint(equalTo: centerXAnchor),
            dateLabel.centerYAnchor.constraint(equalTo: centerYAnchor),
            dateLabel.heightAnchor.constraint(lessThanOrEqualToConstant: 15),
            
            dateBackgroundView.topAnchor.constraint(equalTo: dateLabel.topAnchor, constant: -3),
            dateBackgroundView.leadingAnchor.constraint(equalTo: dateLabel.leadingAnchor, constant: -16),
            dateBackgroundView.bottomAnchor.constraint(equalTo: dateLabel.bottomAnchor, constant: 3),
            dateBackgroundView.trailingAnchor.constraint(equalTo: dateLabel.trailingAnchor, constant: 15),
        ]
        NSLayoutConstraint.activate(constraints)
    }
}
